Intune-PowerShell/DeviceConfiguration/DeviceConfigurationADMX_Export.ps1

```diff
@@ -526,87 +527,32 @@ foreach ($DCP in $DCPs)
-      }
-   ],
-   "definition@odata.bind":"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$($DefinitionValuedefinitionID)')"
-}
-"@
-				
-				$jsonCodePLACEHOLDER = $JSON_Output.Substring(1)
-				$jsonCodePLACEHOLDER = $jsonCodePLACEHOLDER.Replace("<!PLACEHOLDER!>", $_).Substring(0, $jsonCodePLACEHOLDER.Length - 1)
-				$jsonCode = $jsonCode.Replace("<!PLACEHOLDER!>", $jsonCodePLACEHOLDER)
-				write-host "Exporting setting $($DefinitionValuedefinitionDisplayName) to folder $ExportPath\$($FolderName)\$FileName.json" -ForegroundColor Yellow
-				New-Item "$ExportPath\$($FolderName)\$FileName.json" -ItemType File -Force
-				$jsonCode | Set-Content -LiteralPath "$ExportPath\$($FolderName)\$FileName.json" -Force
-			}
-		}
-		else
-		{
-			
-			$jsonCode = @"
-{
-   "enabled":$($GroupPolicyConfigurationsDefinitionValue.enabled.tostring().Replace("True", "true").Replace("False", "false")),
-    "definition@odata.bind":"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$($DefinitionValuedefinitionID)')"
-}
-"@
-			write-host "Exporting setting $($DefinitionValuedefinitionDisplayName) to folder $ExportPath\$($FolderName)\$FileName.json" -ForegroundColor Yellow
-			New-Item "$ExportPath\$($FolderName)\$FileName.json" -ItemType File -Force
-			$jsonCode | Out-File "$ExportPath\$($FolderName)\$FileName.json" -Encoding ascii -Force
-		}
-		
+		$DefinitionValuePresentationValues = Get-GroupPolicyConfigurationsDefinitionValuesPresentationValues -GroupPolicyConfigurationID $DCP.id -GroupPolicyConfigurationsDefinitionValueID $GroupPolicyConfigurationsDefinitionValue.id
+		$OutDef = New-Object -TypeName PSCustomObject
+        $OutDef | Add-Member -MemberType NoteProperty -Name "definition@odata.bind" -Value "https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$definitionValuedefinitionID')"
+        $OutDef | Add-Member -MemberType NoteProperty -Name "enabled" -value $($GroupPolicyConfigurationsDefinitionValue.enabled.tostring().tolower())
+        if ($DefinitionValuePresentationValues.count -gt 1) {
+            $i = 0
+            $PresValues = @()
+            foreach ($Pres in $DefinitionValuePresentationValues) {
+                $P = $pres | Select-Object -Property * -ExcludeProperty id, createdDateTime, lastModifiedDateTime, version
+                $GPDPID = $groupPolicyDefinitionsPresentations[$i].id
+                $P | Add-Member -MemberType NoteProperty -Name "presentation@odata.bind" -Value "https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$definitionValuedefinitionID')/presentations('$GPDPID')"
+                $PresValues += $P
+                $i++
+            }
+            $OutDef | Add-Member -MemberType NoteProperty -Name "presentationValues" -Value $PresValues
+        }
+		$FileName = $definitionValuedefinitionDisplayName -replace '\<|\>|:|"|/|\\|\||\?|\*', "_"
```

Comment:
when settings has same name, this overwrite the existing file.

Reply:
Good point, 
we will refactor to use maybe the definition ID, the only thing is then it wont be human readable, 
I'm not on my prime laptop at the moment so can't test, but we might even be able to use the full variable path to handle that, we will reply with an updated version in the next day or so :)

Reply:
I've reviewed the script, and line 553 should be updated to this:

$FileName = "$($DefinitionValuedefinition.categoryPath)-$definitionValuedefinitionDisplayName" -replace '\<|\>|:|"|/|\\|\||\?|\*', "_"

as it will include the full path for the ADMX setting in front of each of the actual settings.

Reply:
cool, thanks @onpremcloudguy - i've updated the code with your change. I've run it and it's doing what it should. @sandytsang - let me know if there's anything else required

Reply:
Hey guys, thanks a lot for helping out fix this script! 
I changed the file name to 
$FileName = (Join-Path $DefinitionValuedefinition.categoryPath $($definitionValuedefinitionDisplayName)) -replace '\<|\>|:|"|/|\\|\||\?|\*', "_"

And I had change line 541 to if ($DefinitionValuePresentationValues), because if ($DefinitionValuePresentationValues -gt 1) didn't work in my test. 

Now I tested export and import, both works perfect.